velox/functions/sparksql/Arithmetic.h

```diff
@@ -399,7 +398,7 @@ struct UnHexFunction {
       if (first == -1 || second == -1) {
         return false;
       }
-      resultBuffer[i / 2] = (first << 4) | second;
+      resultBuffer[i / 2 + oddShift] = (first << 4) | second;
```

Comment:
Seems we can remove variable `oddShift` and just use `resultBuffer[(i + 1) / 2] = ...`.

Reply:
Another option might be to move `resultBuffer` by offset, and  the variable `i` could be removed.